base/abstractdict.jl

```diff
@@ -335,6 +335,75 @@ function _typeddict(d::AbstractDict, others::AbstractDict...)
     Dict{K,V}(d)
 end
 
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return _merge!(empty(first(dicts)),dicts) # use `empty`. `similar` makes routine 20x slower
+end
+
+function reduce(::typeof(merge), d::T, dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && return copy(d)
+    return _merge!(empty(first(dicts)),d,dicts) # There is no evidence that calling _merge! twice is less efficient.
+end
+
+# This catches typeof(T) == UnionAll.
+# The conversion of `dicts` to a vector of type Any makes this method much faster.
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, Any[x for x in dicts])
+end
+
+function reduce(::typeof(merge), d::AbstractDict, dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, d, Any[x for x in dicts])
+end
+
+function reduce(m::typeof(merge), items::AbstractVector)
+    isempty(items) && throw(ArgumentError("Cannot merge empty vector"))
+    length(items) == 1 && return copy(first(items))
```

Comment:
` reduce(merge, [x])` currently returns `x` without copying, so this isn't needed: you can pass an empty array when there's a single element.

Same below.

Reply:
I don't understand `empty array`.  To agree with the current behavior, we could return `first(items)`. But, you are saying something else ?

Reply:
I mean that `reduce(merge, items)` is equivalent to `first(items)` when there's only one element, so no need to add special code for that case.

Reply:
Ok. Got it. Simply remove the line and there is no observable difference. You are arguing again for simplicity over efficiency.